File: Leetcode/TwoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
       dic = {}
       for i, num in enumerate(nums):
           if target - num in dic:
               return [dic[target - num], i]
           dic[num] = i           
if __name__ == "__main__":
    s = Solution()
    print(s.twoSum([2,7,11,15], 9)) # [0, 1]
    print(s.twoSum([3,2,4], 6)) # [1, 2]
    print(s.twoSum([3,3], 6)) # [0, 1]

File: Leetcode/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        vector<int> sumOf(nums.size());

        for (int i = 0; i < nums.size(); i++)
            for (int j = i + 1; j <= nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    sumOf.push_back(i);
                    sumOf.push_back(j);
                    return sumOf;
                }
            }
        return sumOf;
    }
};